ckanext/unaids/logic.py

```diff
@@ -35,3 +36,49 @@ def update_filename_in_resource_url(resource):
         new_url_segments = url_segments[:-1] + [filename]
         resource['url'] = '/'.join(new_url_segments)
     return resource
+
+
+def auto_populate_data_dictionary(context, resource_dict):
+    table_schema_name = resource_dict.get('schema')
+
+    if not table_schema_name:
+        return
+
+    table_schema = validation_load_json_schema(table_schema_name)
+
+    if not table_schema:
+        raise toolkit.ObjectNotFound(
+            'Resource table schema "{}" does not exist'.format(table_schema)
+        )
+
+    field_schemas = {field['name']: field for field in table_schema.get('fields')}
```

Comment:
```
{field['name']: field for field in table_schema.get('fields', [])}
```
default value empty list instead of `None`. Otherwise you'll get:
```
TypeError: 'NoneType' object is not iterable
```